PROTOCOL.md

```diff
@@ -0,0 +1,99 @@
+# GraphQL over WebSocket Protocol
+
+## Client-server communication
+
+Each message has a `type` field, which defined in the protocol of this package, as well as associated fields inside `payload` field, depending on the message type, and `id` field so the client can identify each response from the server.
+
+The structure of each client side message is:
+```typescript
+export interface OperationMessage {
+  payload?: any;
+  id?: string;
+  type: string;
+}
+```
+
+### Client -> Server
+
+#### GQL_CONNECTION_INIT
+Client sends this message after plain websocket connection to start the communication with the server
+
+This message also triggers the `onConnect` callback on the server.
+
+- `payload: Object` : optional parameters that the client specifies in `connectionParams`
+
+#### GQL_CONNECTION_TERMINATE
+Client sends this message to terminate the connection.
+
+#### GQL_START
+Client sends this message to execute GraphQL operation
+- `id: string` : The id of the GraphQL operation to start
+- `payload: Object`:
+    * `query: string` : GraphQL operation as string or parsed GraphQL document node
+    * `variables?: Object` : Object with GraphQL variables
+    * `operationName?: string` : GraphQL operation name
+    * `context?: any` : Execution context for the operation
+    
+#### GQL_STOP
+Client sends this message in order to stop a running GraphQL operation execution (for example: unsubscribe)
+- `id: string` : operation id
+    
+### Server -> Client
+
+#### GQL_CONNECTION_ERROR
```

Comment:
Missing info: the server should reject messages sent to it before connection was completed.